src/mscorlib/shared/System/Double.cs

```diff
@@ -221,16 +223,18 @@ public bool Equals(Double obj)
         //The hashcode for a double is the absolute value of the integer representation
         //of that double.
         //
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // 64-bit constants make the IL unusually large that makes the inliner to reject the method
         public unsafe override int GetHashCode()
         {
-            double d = m_value;
-            if (d == 0)
+            var bits = Unsafe.As<double, long>(ref m_value);
```

Comment:
I guess you could drop "unsafe" from the method def now that `Unsafe.As` is used.